scripts/linkbot/monitor-raspbian-os.js

```diff
@@ -70,46 +21,40 @@ function updateGitlabCILink(newLine, newURL, newName) {
     console.error(err);
     process.exit(1);
   }
-}
-
-JSDOM.fromURL(url, {}).then((dom) => {
-  var document = dom.window.document;
-  var table = document.getElementsByTagName("table");
-  var rows = table[0].rows;
-  var matches = [];
-  for (var i = 0; i < rows.length; i++) {
-    var rowText = rows[i].textContent;
-    var regMatch = rowText.match(reg);
-    if (regMatch) {
-      matches.push(regMatch);
-    }
-  }
-  // Sort the accumulated matches
-  matches.sort(function (a, b) {
-    let al = Date.parse(a.groups.date);
-    let bl = Date.parse(b.groups.date);
-    if (al == bl) {
-      let ad = Date.parse(a.groups.updated);
-      let bd = Date.parse(b.groups.updated);
-      return bd - ad;
+};
+
+const checkForUpdates = async (url, currentImageName) =>
+  getLinksByMatch(url, reg).then(async (links) => {
+    const { link, match } = links.sort((a, b) => Date.parse(b.match.groups.date) - Date.parse(a.match.groups.date))[0];
```

Comment:
tbh, I was hoping you were gonna show me some magic here.

I remember when I wrote this, that it felt like some abstraction was escaping me here, and there'd be som elegant way of forging this, considering it's pretty much similar in a lot of the updaters.

Reply:
the date based comparisons are similar in places, but still dependent on the regex groups or where the date is gotten from, so still somewhat unique each monitor... IMO if the versioning scheme was the same for all the images that would be a more useful abstraction, but here we compare dates in one place and ubuntu version strings in another and then again something more semver like... that is to say: I don't see a better abstraction either unfortunately..

Reply:
:'(